SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -30,71 +32,27 @@ DEFINE_LOG_CATEGORY(LogSpatialDeploymentManager);
 
 #define LOCTEXT_NAMESPACE "FLocalDeploymentManager"
 
-static const FString SpatialServiceVersion(TEXT("20200611.170527.924b1f1c45"));
-
 FLocalDeploymentManager::FLocalDeploymentManager()
 	: bLocalDeploymentRunning(false)
-	, bSpatialServiceRunning(false)
-	, bSpatialServiceInProjectDirectory(false)
 	, bStartingDeployment(false)
 	, bStoppingDeployment(false)
-	, bStartingSpatialService(false)
-	, bStoppingSpatialService(false)
 {
 }
 
 void FLocalDeploymentManager::PreInit(bool bChinaEnabled)
 {
-	bIsInChina = bChinaEnabled;
-	// Don't kick off background processes when running commandlets
-	const bool bCommandletRunning = IsRunningCommandlet();
-
-	// Check for the existence of Spatial and Spot. If they don't exist then don't start any background processes.
-	const bool bSpatialServicesAvailable = FSpatialGDKServicesModule::SpatialPreRunChecks(bIsInChina);
-
-	if (bCommandletRunning || !bSpatialServicesAvailable)
-	{
-		if (!bSpatialServicesAvailable)
-		{
-			UE_LOG(LogSpatialDeploymentManager, Warning,
-				   TEXT("Pre run checks for LocalDeploymentManager failed. Local deployments cannot be started."));
-		}
-		bLocalDeploymentManagerEnabled = false;
-		return;
-	}
-
 	// Ensure the worker.jsons are up to date.
 	WorkerBuildConfigAsync();
 
 	// Watch the worker config directory for changes.
 	StartUpWorkerConfigDirectoryWatcher();
 }
 
-void FLocalDeploymentManager::Init(FString RuntimeIPToExpose)
+void FLocalDeploymentManager::Init()
 {
-	if (bLocalDeploymentManagerEnabled)
-	{
-		// If a service was running, restart to guarantee that the service is running in this project with the correct settings.
-		UE_LOG(LogSpatialDeploymentManager, Log, TEXT("(Re)starting Spatial service in this project."));
-
-		AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this, RuntimeIPToExpose] {
-			// Stop existing spatial service to guarantee that any new existing spatial service would be running in the current project.
-			TryStopSpatialService();
-			// Start spatial service in the current project if spatial networking is enabled
-
-			if (GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking())
-			{
-				TryStartSpatialService(RuntimeIPToExpose);
-			}
-			else
-			{
-				UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("SpatialOS daemon not started because spatial networking is disabled."));
-			}
-
-			// Ensure we have an up to date state of the spatial service and local deployment.
-			RefreshServiceStatus();
-		});
-	}
+	// Kill any existing runtime processes.
+	// We cannot attach to old runtime processes as they may be 'zombie' and not killable (even if they are not blocking ports).
```

Comment:
how do they become zombie? is there any way to stop that?

Reply:
This is a windows issue unfortunately and I couldn't get to the bottom of it, seems to be well known online. Essentially the process is in a stopped state where it's not consuming any CPU cycles or blocking any ports, but it still has a PID and can be looked at, it also cannot be destroyed for some reason. This means that we can't realistically attach to already running found processes. I tried various means with no success. The cleanest approach is to attempt to kill any existing runtime and then launch a new one where we can ensure it's a valid process and we can own the lifetime for this session.

Reply:
Is there any link that explains the issue that you could add to the comment?

Reply:
Sure can do, one of:
https://stackoverflow.com/questions/49988/really-killing-a-process-in-windows
https://superuser.com/questions/752435/how-to-get-rid-of-zombie-processes-without-parents-in-windows-8